server/docs/getting-started.md

```diff
@@ -2,106 +2,251 @@
+One more thing: to streamline your first dapp, we've built and deployed a bot that will accept `proposeInstallVirtual()` requests for **HighRoller** and play the game with your user. This means you only have to code the UI for the first player of **HighRoller**.
 
-2. **State management in a channel** - In an open channel, requests to modify state are passed to the CF node (via connection 1). The node calls the pure functions of the solidity contract (via connection 2) to verify that requests to modify state are valid, and to alter the state accordingly. The node makes updated state available to members of the channel (via connection 1).
 
-3. **Ending a game**  - When the game is over, the dapp (HighRoller.js) must request (via connection 1) that the node end the game. In turn, the CF node will (via connection 2) verify that the game is over, and if it is over, have the contract implement the transactions that resolve the game.
 
+---------
 
-### In this Getting Start Guide, you’ll learn how to:
+## A Quick Look at the Contract
+
+Let's take a quick look at the [HighRoller.sol](https://github.com/counterfactual/monorepo/blob/master/packages/apps/contracts/HighRollerApp.sol) contract to see how the game logic is structured.
+
+### Game logic for HighRoller.sol
+
+We want to make our dice game very secure: the application logic requires both users to submit a random number -- the product of the two numbers is used as the source of randomness for both players' dice rolls; this is what makes it secure. In order to prevent the second player from using information about the first player's number, we use a commit-by-hash and reveal paradigm:
+  * The first player submits a **hash** of their **number** with a random **salt**
+  * The second player submits their **number**
+  * The first player **reveals** their number by submitting the number and the salt, which are checked against the original hash
+  * The contract generates die rolls for each player
+
+The information above is enough to determine the structure of the **HighRoller** `AppState` and `Action`, both of which are required for Counterfactual Apps. We also note the data structures `ActionType` and `Stage` that the contract developer chose to create:
+
+```solidity
+contract HighRollerApp is CounterfactualApp {
+
+  enum ActionType {
+    START_GAME,
+    COMMIT_TO_HASH,
+    COMMIT_TO_NUM,
+    REVEAL
+  }
+
+  enum Stage {
+    PRE_GAME,
+    COMMITTING_HASH,
+    COMMITTING_NUM,
+    REVEALING,
+    DONE
+  }
+
+  enum Player {
+    FIRST,
+    SECOND
+  }
+
+  struct AppState {
+    address[2] playerAddrs;
+    Stage stage;
+    bytes32 salt;
+    bytes32 commitHash;
+    uint256 playerFirstNumber;
+    uint256 playerSecondNumber;
+  }
+
+  struct Action {
+    ActionType actionType;
+    uint256 number;
+    bytes32 actionHash;
+  }
+```
+
+Because it’s such a simple game (Player1 → Player2 → Player1 → DONE), `getTurnTaker()` and `isStateTerminal()` are easy to implement:
```

Comment:
I think you can describe the game / state machine in a sentence.

"The application is quite simple. Player 1 commits to a preimage of a secret, Player 2 chooses a number, Player 1 reveals their secret, Player 2 computes the product and determines the dice roll and the game is over. This simple flow is represented in the implementation of the `getTurnTaker` and `isStateTerminal` methods."

I would omit the `()` in the method names because it makes it seem like there are no arguments.